File: Core/src/main/java/com/cyrillrx/android/ws/SimpleLoaderAdapter.java
package com.cyrillrx.android.ws;

import android.support.v7.widget.RecyclerView;

import com.cyrillrx.android.section.ItemType;

/**
 * @author Cyril Leroux
 *         Created on 17/03/15
 */
public abstract class SimpleLoaderAdapter<VH extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<VH> {

    protected boolean isLoading;

    public abstract void addLoader();

    public abstract void removeLoader();

    public boolean isLoader(int position) { return getItemViewType(position) == ItemType.LOADER; }
}

File: Core/src/main/java/com/cyrillrx/android/ws/SimpleLoaderAdapter.kt
package com.cyrillrx.android.ws

import android.support.v7.widget.RecyclerView

import com.cyrillrx.android.section.ItemType

/**
 * @author Cyril Leroux
 * *         Created on 17/03/15
 */
abstract class SimpleLoaderAdapter<VH : RecyclerView.ViewHolder> : RecyclerView.Adapter<VH>() {

    protected var isLoading: Boolean = false

    abstract fun addLoader()

    abstract fun removeLoader()

    fun isLoader(position: Int): Boolean {
        return getItemViewType(position) == ItemType.LOADER
    }
}
